controllers/clustergroupupgrade_controller.go

```diff
@@ -147,61 +153,35 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 				nextReconcile = ctrl.Result{RequeueAfter: requeueAfter}
 			}
 
-			var isBatchComplete, isBatchPlatformUpgradeComplete, isBatchOperatorUpgradeComplete bool
+			var isBatchComplete bool
 
-			platformUpgradePolicy, err := r.getBatchPlatformUpgradePolicy(ctx, clusterGroupUpgrade)
+			// At first, assume all clusters in the batch start applying policies starting with the first one.
+			r.initializeRemediationPolicyForBatch(clusterGroupUpgrade)
+			r.Log.Info("\n\n Checking batch for new remediation policies ", "batch index", clusterGroupUpgrade.Status.Status.CurrentBatch)
+			// Check if current policies have become compliant and if new policies have to be applied.
+			err, isBatchComplete := r.getNextRemediationPoliciesForBatch(ctx, clusterGroupUpgrade)
 			if err != nil {
 				return ctrl.Result{}, err
 			}
-			if platformUpgradePolicy != nil {
-				err = r.ensureBatchPlatformUpgradePolicyIsEnforced(ctx, clusterGroupUpgrade, platformUpgradePolicy)
-				if err != nil {
-					return ctrl.Result{}, err
-				}
 
-				isBatchPlatformUpgradeComplete, err = r.isBatchPlatformUpgradePolicyCompliant(ctx, clusterGroupUpgrade)
-				if err != nil {
-					return ctrl.Result{}, err
-				}
-			}
-
-			operatorUpgradePolicy, err := r.getBatchOperatorUpgradePolicy(ctx, clusterGroupUpgrade)
+			// Add the needed cluster names to upgrade to the appropriate placement rule.
+			err = r.addClustersToPlacementRule(ctx, clusterGroupUpgrade)
```

Comment:
I see your point. My approach was to remove all clusters from the placementRules once the batch is done upgrading. The reason I like this approach is because technically, when a batch is done upgrading, all the copied policies are compliant, if we ever want to check. I think it would be nice to have this option to check in the future. What do you think?

Is it ok if I document the current functionality in a method level comment, as we discussed in the comments right above?

Reply:
That sounds great. I also agree that having all the policies showing compliant at the end of the batch is a good thing.

Reply:
Done - added method comment.